slang-gfx.h

```diff
@@ -146,6 +172,12 @@ enum class Format
     CountOf,
 };
 
+struct FormatInfo
+{
+    uint8_t channelCount;       ///< The amount of channels in the format. Only set if the channelType is set 
+    uint8_t channelType;        ///< One of SlangScalarType None if type isn't made up of elements of type.
```

Comment:
Why isn't this declared as a `SlangScalarType` if that's what it represents? It seems like packing this all into 16 bits might be over-optimizing?

Reply:
If we had a C++ sized type then this would be a no brainer - it could be a byte and type safe. That it is just a typedef to an int without any type safety makes the above choice less problematic. 
That said there is some small readability/lookup benefit perhaps in having it's type be SlangScalarType as opposed to reading in the comment.

Reply:
Well, even if we used explicitly-sized `enum` types in the `slang.h` API, I suspect they'd all be 32 bits in size. In fact, many of the C `enum` declarations use a seaprate `typedef` already to make sure they are explicitly-sized (and all those use 32 bits IIRC).

I think I'm just far more concerned with the stability of binary interfaces than with making things really tightly bit-packed. I'm already nervous about stuff like our texture "shape" and "access" bitfield where we might be in danger of running out of bits. I think my mind is in "better safe than sorry" mode.